pkgs/by-name/xo/xonsh/package.nix

```diff
@@ -1,23 +1,122 @@
-{ lib
-, callPackage
-, extraPackages ? (ps: [ ])
-, runCommand
+{
+  lib,
+  callPackage,
+  coreutils,
+  fetchFromGitHub,
+  fetchPypi,
+  git,
+  gitUpdater,
+  glibcLocales,
+  python3,
 }:
 
 let
-  xonsh-unwrapped = callPackage ./unwrapped.nix { };
-  inherit (xonsh-unwrapped.passthru) python;
+  # workaround for "Bad file descriptor in prompt_toolkit > 3.0.40"
+  # https://github.com/xonsh/xonsh/issues/5241
+  # https://github.com/prompt-toolkit/python-prompt-toolkit/issues/1853
+  prompt-toolkit' = python3.pkgs.prompt-toolkit.overridePythonAttrs (old: {
+    version = "3.0.40";
```

Comment:
Xonsh just [rolled back](https://github.com/xonsh/xonsh/pull/5403) to prompt-toolkit 3.0.39, since the still similar issues popping up with 3.0.40.